src/components/molecules/ChatsList/ChatsList.tsx

```diff
@@ -0,0 +1,179 @@
+    chatUsers: state.firestore.data.chatUsers,
+  }));
+
+  const [selectedUser, setSelectedUser] = useState<WithId<User>>();
+
+  const discussionPartnerWithLastMessageExchanged =
+    privateChats &&
+    privateChats.reduce<LastMessageByUser>((agg, item) => {
+      let lastMessageTimeStamp;
+      let discussionPartner;
+      if (item.from === user?.uid) {
+        discussionPartner = item.to;
+      } else {
+        discussionPartner = item.from;
+      }
+      try {
+        lastMessageTimeStamp = agg[discussionPartner].ts_utc;
+        if (lastMessageTimeStamp < item.ts_utc) {
+          agg[discussionPartner] = item;
+        }
+      } catch {
+        agg[discussionPartner] = item;
+      }
+      return agg;
+    }, {});
+
+  const onClickOnSender = (sender: WithId<User>) => {
+    const chatsToUpdate = privateChats.filter(
+      (chat) => !chat.isRead && chat.from === sender.id
+    );
+    chatsToUpdate.map(
+      (chat) => user && setPrivateChatMessageIsRead(user.uid, chat.id)
+    );
+    setSelectedUser(sender);
+  };
+
+  const roundToNearestHour = (seconds: number) => {
+    const oneHour = 60 * 60;
+    return Math.floor(seconds / oneHour) * oneHour;
+  };
+  const DAYS_AGO = getDaysAgoInSeconds(VENUE_CHAT_AGE_DAYS);
+  const HIDE_BEFORE = roundToNearestHour(DAYS_AGO);
+
+  const chatsToDisplay = useMemo(
+    () =>
+      privateChats &&
+      privateChats
+        .filter(
+          (message) =>
+            message.deleted !== true &&
+            message.type === "private" &&
+            (message.to === selectedUser?.id ||
+              message.from === selectedUser?.id) &&
+            message.ts_utc.seconds > HIDE_BEFORE
+        )
+        .sort((a, b) => b.ts_utc.valueOf().localeCompare(a.ts_utc.valueOf())),
+    [privateChats, selectedUser, HIDE_BEFORE]
+  );
+
+  const chatContext = useContext(ChatContext);
+  const onMessageSubmit = async (data: { messageToTheBand: string }) => {
+    chatContext &&
+      user &&
+      chatContext.sendPrivateChat(
+        user.uid,
+        selectedUser!.id,
+        data.messageToTheBand
+      );
+  };
+
+  const hasPrivateChats = !isEmpty(discussionPartnerWithLastMessageExchanged);
+
+  return selectedUser ? (
+    <>
+      <div
+        className="private-container-back-btn"
+        onClick={() => setSelectedUser(undefined)}
+      />
+      <ChatBox chats={chatsToDisplay} onMessageSubmit={onMessageSubmit} />
+    </>
+  ) : (
+    <>
+      {hasPrivateChats && (
+        <div className="private-container show">
+          <div className="private-messages-list">
+            {Object.keys(discussionPartnerWithLastMessageExchanged)
+              .sort((a, b) =>
+                discussionPartnerWithLastMessageExchanged[b].ts_utc
+                  .valueOf()
+                  .localeCompare(
+                    discussionPartnerWithLastMessageExchanged[
+                      a
+                    ].ts_utc.valueOf()
+                  )
+              )
```

Comment:
We should extract the `Object.keys` and `.sort` from the JSX, into a `const`/similar above, and similarly can extract the `sort` function into a static `const` outside of the component.

Will probably also want to `useMemo` the result so we don't process it on every render.

Reply:
fixed in 77fc5d6d1829a42da2152d00871cde0b6f47269d